### ISSUE TITLE
no-inner-declarations wrongfully fails on exports after the first statement

### ISSUE THREAD

**Issue by RDambrosio016** [RDambrosio016/RSLint#12]
**I tried this code**:

```js
export const foo = [];
export function bar() {}
```

**I expected this to happen**: *linting passes*

**Instead this happened:**: 

```
error[no-inner-declarations]: move this function declaration to program's root
  ┌─ tests\main.mjs:2:8
  │
2 │ export function bar() {}
  │        ^^^^^^^^^^^^^^^^^
  │
  = help: move the declaration to the program root
```
### Meta 
`rslint -V`: `rslint 0.1.0`

If building from source:
`rustc --version --verbose`:
```
rustc 1.48.0-nightly (f68e08933 2020-09-19)
binary: rustc
commit-hash: f68e08933d8f519a9655934fedebbc509661b219
commit-date: 2020-09-19
host: x86_64-pc-windows-msvc
release: 1.48.0-nightly
LLVM version: 11.0
```

**If 
**Additional context**

**Comment by Stupremee** [comment]
If `rslint_parse` parses the file as a module, the linting passes. This is caused because in your code (if its parsed as a script), the `parent.kind()` is `ERROR` and thus will be reported. This could be easily fixed by just checking for `ERROR` in the lint.

**Comment by RDambrosio016** [comment]
Oh that is true, and this brings up another point, the linter should not tread into ERROR nodes, it should skip them and their children because the data inside is non-sensical

**Comment by Stupremee** [comment]
This is going to be complicated, because many lints use `.parent()` or `.children()`, which are not guaranteed to not return `ERROR`.

**Comment by RDambrosio016** [comment]
that does not matter if the linter does not step into the children of error nodes

**Comment by Stupremee** [comment]
Ah yes, that makes sense. Misunderstood your message

**Comment by RDambrosio016** [comment]
Although the file is `.mjs` so it should be parsing as module 🤔

**Comment by Stupremee** [comment]
If I rename it to `.mjs` and run it, the linting passes

**Comment by RDambrosio016** [comment]
haha so the issue is still there, although we caught another issue for ERROR nodes

**Comment by RDambrosio016** [comment]
Oh haha, that makes sense, its an easy fix

**Comment by RDambrosio016** [comment]
Fixed in 970df0382be6d03024489fc888cef2cc29244a5c
